esphome/components/mpr121/mpr121.h

```diff
@@ -49,21 +52,37 @@ class MPR121Channel : public binary_sensor::BinarySensor {
  public:
   void set_channel(uint8_t channel) { channel_ = channel; }
   void process(uint16_t data) { this->publish_state(static_cast<bool>(data & (1 << this->channel_))); }
+  int get_channel() { return this->channel_; };
+  void set_touch_threshold(uint8_t touch_threshold) { this->touch_threshold_ = touch_threshold; };
+  void set_release_threshold(uint8_t release_threshold) { this->release_threshold_ = release_threshold; };
+  uint8_t get_touch_threshold() { return this->touch_threshold_; };
+  uint8_t get_release_threshold() { return this->release_threshold_; };
 
  protected:
   uint8_t channel_{0};
+  uint8_t touch_threshold_{DEFAULT_TOUCH_THRESHOLD};
```

Comment:
Ok I understand the optional threshold problem now a bit more.

Though I'm not so happy about the implementation here tbh. ESPHome has a powerful type for this: `optional`.

Change this to:

```suggestion
  optional<uint8_t> touch_threshold_{};
```

and in the `for (auto *channel : this->channels_) {` loop, use 

```cpp
channel->get_touch_threshold().value_or(this->touch_threshold_);
```

Reply:
I made `MPR121Component` a friend class of `MPR121Channel`, this way I could skip the getters and access the `MPR121Channel` protected threshold variables.

mpr121.h
```cpp
class MPR121Channel : public binary_sensor::BinarySensor {
  friend class MPR121Component;

 public:
  void set_channel(uint8_t channel) { channel_ = channel; }
  void process(uint16_t data) { this->publish_state(static_cast<bool>(data & (1 << this->channel_))); }
  void set_touch_threshold(uint8_t touch_threshold) { this->touch_threshold_ = touch_threshold; };
  void set_release_threshold(uint8_t release_threshold) { this->release_threshold_ = release_threshold; };

 protected:
  uint8_t channel_{0};
  optional<uint8_t> touch_threshold_{};
  optional<uint8_t> release_threshold_{};
};
```
mpr121.cpp
```cpp
  for (auto *channel : this->channels_) {
    this->write_byte(MPR121_TOUCHTH_0 + 2 * channel->channel_,
                     channel->touch_threshold_.value_or(this->touch_threshold_));
    this->write_byte(MPR121_RELEASETH_0 + 2 * channel->channel_,
                     channel->release_threshold_.value_or(this->release_threshold_));
  }
```